Objective-C/Tests/URLEndpointListenerTest.m

```diff
@@ -437,4 +437,202 @@ - (void) testUnavailableNetworkInterface {
+                   type: kCBLReplicatorTypePushAndPull
+             continuous: NO
+          authenticator: nil
+             serverCert: (__bridge SecCertificateRef) listener1.config.tlsIdentity.certs[0]
+              errorCode: 0
+            errorDomain: nil];
+    
+    [listener1 stop];
+    [listener2 stop];
+    
+    AssertEqual(self.otherDB.count, 1);
+}
+
+- (void) testMultipleReplicatorsOnSameListener {
+    XCTestExpectation* exp1 = [self expectationWithDescription: @"replicator#1 stopped"];
+    XCTestExpectation* exp2 = [self expectationWithDescription: @"replicator#2 stopped"];
+    [self listen]; // writable listener
+    
+    // save a doc on listenerDB
+    NSError* err = nil;
+    CBLMutableDocument* doc = [self createDocument: @"doc"];
+    [doc setValue: @"Tiger" forKey: @"species"];
+    [self.otherDB saveDocument: doc error: &err];
+    AssertNil(err);
+    
+    // save blob on db#1
+    CBLMutableDocument* doc1 = [self createDocument: @"doc-1"];
+    NSData* content = [@"i am a blob" dataUsingEncoding: NSUTF8StringEncoding];
+    CBLBlob* blob = [[CBLBlob alloc] initWithContentType: @"text/plain" data: content];
+    [doc1 setValue: blob forKey: @"blob"];
+    [self saveDocument: doc1];
+    
+    // save blob on db#2
+    [self deleteDBNamed: @"db2" error: &err];
+    AssertNil(err);
+    CBLDatabase* db2 = [self openDBNamed: @"db2" error: &err];
+    AssertNil(err);
+    CBLMutableDocument* doc2 = [self createDocument: @"doc-2"];
+    CBLBlob* blob2 = [[CBLBlob alloc] initWithContentType: @"text/plain" data: content];
+    [doc2 setValue: blob2 forKey: @"blob"];
+    [db2 saveDocument: doc2 error: &err];
+    AssertNil(err);
+    
+    // replicator#1
+    CBLReplicatorConfiguration* config1 = [self configWithTarget: _listener.localEndpoint
+                                                            type: kCBLReplicatorTypePushAndPull
+                                                      continuous: NO];
+    config1.pinnedServerCertificate = (__bridge SecCertificateRef) _listener.config.tlsIdentity.certs[0];
+    CBLReplicator* repl1 = [[CBLReplicator alloc] initWithConfig: config1];
+    
+    // replicator#2
+    CBLReplicatorConfiguration* config2 = [[CBLReplicatorConfiguration alloc] initWithDatabase: db2
+                                                                                        target: _listener.localEndpoint];
+    config2.pinnedServerCertificate = (__bridge SecCertificateRef) _listener.config.tlsIdentity.certs[0];
+    CBLReplicator* repl2 = [[CBLReplicator alloc] initWithConfig: config2];
+    
+    // get listener status
+    __block Listener* weakListener = _listener;
+    __block uint64_t maxConnectionCount = 0, maxActiveCount = 0;
+    id changeListener = ^(CBLReplicatorChange * change) {
+        Listener* listener = weakListener;
+        if (change.status.activity == kCBLReplicatorBusy) {
+            maxConnectionCount = MAX(listener.status.connectionCount, maxConnectionCount);
```

Comment:
There could be a race condition when testing with multiple single shot replicators. For example, if one replicator stops very fast before the other one is busy, the maxConnectionCount could be just 1. Testing with continuous replicators will help on this.

Reply:
to avoid the race, we are passing in a blob. which should at least take some time to finish.

Reply:
If blob is big enough, it would help reducing or avoid the race. If you decided to use blob, please add some comments to the code (in case the test intermittently fails in the future).

Wondering if the push/pull filter can be used as a workaround to sync b/w two replicators so that we can really  ensure that both of them alive at the same time.